### PR TITLE
Add a check for a fake access token on the state to use a complete page reload and prevent blocking from a SMART app launched against an open endpoint FHIR server.

### PR DESCRIPTION
**Pull request by zplata**
Issue: https://github.com/smart-on-fhir/client-js/issues/16

Currently, with the new refresh token support, if there is an app (using this library) that is launched against an open FHIR server endpoint, the new check on the 'exp' payload will still try to execute and evaluate a time on the expiration. This causes a block because the jwt object that decodes the access token may be undefined, thus trying to read an 'exp' property on this object would cause the launch flow to error out.

To remedy this, we should check if the state has a property set during the `oauth` method, `fake_token_response`, which is set if the app is not going through a secured fhir server. If this property exists, we should complete a page reload rather than going through the refresh token flow. This will ensure apps are not stalled due to a reference to an undefined jwt object, in this case, for the open fhir server connection.

### CODE REVIEW

**Comment by zplata** [comment]
@jmandel @nschwertner would either of you be able to take a look at this change?

File: src/client/bb-client.js [hunk 1]
```diff
@@ -262,6 +262,17 @@ function validTokenResponse() {
   return false;
 }
 
+function isFakeOAuthToken(){
+  if (validTokenResponse()) {
+    var token = getPreviousToken();
+    if (token && token.state) {
+      var state = JSON.parse(sessionStorage[token.state]);
```
Comment (jmandel) [inline src/client/bb-client.js]: Doesn't this assume `BBClient.settings.fullSessionStorageSupport = false`? What about the case where it's `true`?

Comment (zplata) [reply src/client/bb-client.js]: I think this should work for that flag when set to `true` or `false`, please let me know if I'm missing something. In either case, the `fake_token_response` property can only be exposed off the state in sessionStorage, whether run against a secured or open FHIR endpoint. When we get the token back, we check session storage for the state value and check the `fake_token_response` property from that once the JSON is parsed.

**Comment by zplata** [comment]
@jmandel As I just discovered with this new flow though, if `fullSessionStorageSupport = false`, we need to accurately check the state parameter in the URL. Currently, this is appended via an anchor symbol (i.e. `#state=abc123`), but was thinking of changing the anchor to a search symbol to change to, i.e. `?state=abc123`. This is because if `fullSessionStorageSupport = false`, when the `isFakeOAuthToken` function is executed, and the `validTokenResponse` function is run inside that, `urlParam` will fail because it cannot correctly parse a url for the state, such as this example, `http://localhost:8080/index.html#state=abc123` and then will proceed with checking `args` instead of using urlParam, which may be undefined.

Let me know your thoughts on this change.

**Comment by jmandel** [comment]
This makes sense to me and sounds like a reasonable change; will you submit in a separate PR?

**Comment by zplata** [comment]
Gladly! It's going through our current fork right now in review, and will PR once it's in.

**Pull request merged**